### PR TITLE
Add more AST utilities for type and return type

### PR DESCRIPTION
**Pull request by neakor**

### CODE REVIEW

File: Sources/SourceParsingFramework/Utilities/ASTUtils.swift [hunk 1]
```diff
@@ -41,6 +56,29 @@ public extension Structure {
         return dictionary["key.name"] as! String
     }
 
+    /// The type of this structure.
+    public var type: ASTType {
+        if let kindString = dictionary["key.kind"] as? String {
+            switch kindString {
+            case "source.lang.swift.decl.class": return .class
```
Comment (rudro) [inline Sources/SourceParsingFramework/Utilities/ASTUtils.swift]: nit: we can avoid typing the strings here by using the ones already in SourceKittenFramework `SwiftDeclarationKind.class.rawValue`

**Review by rudro — APPROVED** [review]
I think it's probably ok to have a `var type` method. However, this is one of those functions, which will run thousands of times (in the event there are thousands of input file, with several subsctructs per file). So I think we should also have a set of methods that avoids the full switch on string and just does a single compare operation. e.g:

`
    var isClass: Bool {
        return kind == SwiftDeclarationKind.class.rawValue
    }
`

File: Sources/SourceParsingFramework/Utilities/ASTUtils.swift (hunk 1 above)
Comment (neakor) [reply Sources/SourceParsingFramework/Utilities/ASTUtils.swift]: Good call. I'll update this.

**Comment by neakor** [comment]
> I think it's probably ok to have a `var type` method. However, this is one of those functions, which will run thousands of times (in the event there are thousands of input file, with several subsctructs per file). So I think we should also have a set of methods that avoids the full switch on string and just does a single compare operation. e.g:
> 
> `var isClass: Bool { return kind == SwiftDeclarationKind.class.rawValue }`

I don't think that level of micro-optimization is really necessary though. Especially with a `switch` we are not running through every single case like `if/else`.

**Review by rudro — APPROVED** [review]
Update looks good.

**Pull request merged**